artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumIntegration/Features/Timeline/GetProfileTimeLineService.cs

```diff
@@ -1,14 +1,69 @@
+using Artificial.Scrum.Master.ScrumIntegration.Exceptions;
+using Artificial.Scrum.Master.ScrumIntegration.Infrastructure.ApiTokens;
+using Artificial.Scrum.Master.ScrumIntegration.Infrastructure.ScrumServiceHttpClient;
+using Artificial.Scrum.Master.ScrumIntegration.Utilities;
+
 namespace Artificial.Scrum.Master.ScrumIntegration.Features.Timeline;
 
 internal interface IGetProfileTimeLineService
 {
-    Task<GetProfileTimeLineResponse> Handle(string userId);
+    Task<List<GetProfileTimeLineResponse>> Handle(string userId);
 }
 
 internal class GetProfileTimeLineService : IGetProfileTimeLineService
 {
-    public Task<GetProfileTimeLineResponse> Handle(string userId)
+    private readonly IUserTokensRepository _userTokensRepository;
+    private readonly IProjectHttpClientWrapper _projectHttpClientWrapper;
+    private readonly IJwtDecoder _jwtDecoder;
+
+    public GetProfileTimeLineService(IUserTokensRepository userTokensRepository,
+        IProjectHttpClientWrapper projectHttpClientWrapper, IJwtDecoder jwtDecoder)
+    {
+        _userTokensRepository = userTokensRepository;
+        _projectHttpClientWrapper = projectHttpClientWrapper;
+        _jwtDecoder = jwtDecoder;
+    }
+
+    public async Task<List<GetProfileTimeLineResponse>> Handle(string userId)
     {
-        throw new NotImplementedException();
+        var userTokens = await _userTokensRepository.GetUserAccessTokens(userId);
```

Comment:
Tak się zastanawiam czy tego całego sprawdzania nie możnaby wynieść do jakiejś klasy IAccessTokenValidator albo IAccessTokenProvider i tam np. metoda ProvideOrThrow(); Ten kod jest kopiowany i z jednej strony według mnie nie jest to jakiś duży problem, bo np. tutaj możemy chcieć inaczej tego tokena walidować ale z drugiej strony tak nie robimy i nie zanosi się żeby tak było, a będzie można to zamockować do testów jednostkowych wtedy i otestować providera osobno, a nie w tej metodzie sprawdzać wyjątki

Reply:
Możesz to zostawić, jest git